nestkernel/conn_builder.cpp

```diff
@@ -341,6 +341,41 @@ nest::ConnBuilder::connect()
       "This connection rule does not support symmetric connections." );
   }
 
+  if ( kernel().model_manager.connector_requires_clopath_archiving(
+         synapse_model_id_ ) )
+  {
+    // get thread id
+    const thread tid = kernel().vp_manager.get_thread_id();
+    for ( GIDCollection::const_iterator tgid = targets_->begin();
+          tgid != targets_->end();
+          ++tgid )
+    {
+      Node* const target = kernel().node_manager.get_node( *tgid, tid );
+      if ( not target->is_proxy() and not target->supports_clopath_archiving() )
```

Comment:
Can this test (`supports_clopath_archiving()`) be replaced by `dynamic_cast<Clopath_Archiving_Node*>(target) != NULL`?

Reply:
I like the idea, changed with 69bfa0d